components/eamxx/src/physics/mam/mam_coupling.hpp

```diff
@@ -238,7 +223,8 @@ const char* cld_aero_mmr_field_name(const int mode, const int species) {
   if (!cld_aero_mmr_names(mode, species)[0]) {
     const auto aero_id = mam4::mode_aero_species(mode, species);
     if (aero_id != mam4::AeroId::None) {
-      concat_3_strings(aero_species_name(static_cast<int>(aero_id)),
+      auto aerosol_species_name =mam4::aero_id_short_name(aero_id);
```

Comment:
@jeff-cohere , I am using `std::string `for this species name return from` mam4::aero_id_short_name`.  It should work. However, I noted you are using `char*` in `concat_3_strings`. Should we switch `std::string ` to `char* in  `mam4::aero_id_short_name`?

Reply:
I think when I wrote this I was just favoring C strings over C++ strings because I wasn't sure exactly where all the code would end up being executed (i.e. on CPU or GPU). Switching to C++ strings is fine for anything that doesn't need to run on the GPU.

Reply:
This part of the code should not be invoked by GPUs. Is this correct @bartgol ?

Reply:
This routine is marked with KOKKOS_INLINE_FUNCTION, so I'm assuming you are planning to call it from device. Maybe the better question is why you need to deal with strings on device? Usually, we address different variables by an integer idx when we're in device (i.e., performance critical) code...

Reply:
We are not planning to call this routine from device. I will removed KOKKOS_INLINE_FUNCTION.

Reply:
@bartgol , what is the routine that you mentioned in your previous comment?

Reply:
It is `cld_aero_mmr_field_name`. @jeff-cohere, we are not planning to invoke this routine from the device. is this correct?

Reply:
It should be possible to avoid calling this function on the GPU, unless we're using it there already and don't realize it. Maybe the easiest thing is to switch it over to using C++ strings, get rid of `KOKKOS_INLINE_FUNCTION`, and see whether things still work.

Reply:
I will remove `KOKKOS_INLINE_FUNCTION` from routines that return  a` char*`.  In  a future PR, we can switch `char*`  to C++ strings.